petastorm/pytorch.py

```diff
@@ -117,15 +150,20 @@ def __iter__(self):
             logger.warning('Start a new pass of Petastorm DataLoader, reset underlying Petastorm reader to position 0.')
         self._in_iter = True
 
+        iterator = self._iter_impl()
         try:
-            for batch in self._iter_impl():
+            if self._queue_size > 1:
+                iterator = BackgroundIterator(iterator, queue_size=self._queue_size)
+            for batch in iterator:
                 yield batch
         except Exception as e:
             self._error = e
             logger.error('Iteration on Petastorm DataLoader raise error: %s', repr(e))
             raise
         finally:
             self._in_iter = False
+            if isinstance(iterator, BackgroundIterator):
+                iterator.stop.set()
```

Comment:
Let's make `stop` a private member (i.e. `_stop`) and add an API to the BackgroundIterator that performs stop (encapsulation principle).